lib/std/net.zig

```diff
@@ -12,12 +12,55 @@ const os = std.os;
 const fs = std.fs;
 const io = std.io;
 
+const windows = os.windows;
+const ws2_32 = os.windows.ws2_32;
+
+const dns = @import("net/dns.zig");
+
+const is_windows = builtin.os.tag == .windows;
+
 // Windows 10 added support for unix sockets in build 17063, redstone 4 is the
 // first release to support them.
 pub const has_unix_sockets = @hasDecl(os, "sockaddr_un") and
     (builtin.os.tag != .windows or
     std.Target.current.os.version_range.windows.isAtLeast(.win10_rs4) orelse false);
 
+// Platform-independent socket descriptor.
+// Note that on some platforms this may not be interchangeable with a regular
+// files descriptor.
+pub const Socket = extern struct {
```

Comment:
This wrapper object around `socket_t` brings up some questions.

* When would someone use `Socket` vs `os.socket_t`?
* How would someone coerce an array of `Socket` into an array of handles or file descriptors on windows/posix?  Can the user make the assumption that `Socket` is always the same alignment/size as a handle/file descriptor on those platforms?  In which case they should use a cast?
* When representing an extern struct that contains a socket handle, should `Socket` or `socket_t` be used?
* Should Zig libraries that provide functions that accept sockets be accepting `socket_t` or `Socket`?

The answers to some of these questions will affect the answers to others.  Having multiple type for a socket seems problematic to me unless there are some good answers to these questions.

In my experience in working with networking libraries in various languages, the easiest ones to work with seem to be those that don't try to hide the fact that a socket is just a handle and/or file descriptor.  Creating a wrapper around socket can make simple examples look "nice" but seems to create a "split" in APIs where each version of a socket is used either out of necessity or happenstance.  Sometimes the fact that a socket is a handle/file descriptor is required knowledge to store/pass them around correctly and reflecting this equivalence in the type system seems to make things simpler.

Reply:
Agree, this `Socket` wrapper is a bit useless since `os.socket_t` already exists